test/supervisor_test.exs

```diff
@@ -75,16 +75,19 @@ defmodule SupervisorTest do
 
   describe ".start_child/2" do
     test "starts a process", context do
-      assert {:ok, _pid} = Horde.Supervisor.start_child(context.horde_1, context.task_def)
+      assert {:ok, pid} = Horde.Supervisor.start_child(context.horde_1, context.task_def)
 
-      assert {:error, {:already_started, _child}} =
-               Horde.Supervisor.start_child(context.horde_1, context.task_def)
+      assert_receive {:process_started, ^pid}
     end
 
     test "returns error when starting same process twice", context do
       assert {:ok, pid} = Horde.Supervisor.start_child(context.horde_1, context.task_def)
 
-      assert_receive {:process_started, ^pid}
+      assert {:error, {:already_started, child}} =
+               Horde.Supervisor.start_child(context.horde_1, context.task_def)
+
+      assert is_pid(child)
```

Comment:
you can clean up this part by asserting on line 86 as follows:

`assert {:error, {:already_started, ^pid}} = ...`

Reply:
Mmmh yes, but no one assures me that pid is really a pid :). What if both returns nil? I'll clean it up but add a type assertion if ok for you.